backend/battle/views.py

```diff
@@ -69,3 +69,9 @@ def get_queryset(self):
 class BattleDetail(DetailView):
     model = Battle
     template_name = "battle/battle_detail.html"
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()  # pylint: disable=attribute-defined-outside-init
+        team = Team.objects.filter(battle=self.object, trainer=self.request.user)
+        context = self.get_context_data(object=self.object, team=team)
+        return self.render_to_response(context)
```

Comment:
Please update the context on the `get_context_data`

Reply:
Okay, thanks!